doc/source/default-settings.rst

```diff
@@ -165,6 +165,41 @@ General settings
 
     **Note** The older (misspelled) version ``"formater"`` is deprecated.
 
+.. papis-config:: slugify-lowercase
+    :type: bool
+
+    This setting controls whether capital letters in a (generated or specified)
+    document path should be lowercased before the path is created.
+
+    **Note:** Lowercasing happens *before* removing disallowed characters.
+    If you disable it without allowing uppercase letters they will simply
+    disappear from document paths!
+
+.. papis-config:: slugify-regex-pattern
+    :type: str
+
+    This setting defines the set of *disallowed* characters for document paths.
+    It must be a valid regular expression matching all disallowed characters.
+    The default value disallows all characters except lowercase letters and
+    numbers (and dots for file extensions).
+
+    Possible alternative values are:
+
+    * ``r"[^a-zA-Z0-9.]+"``: Don't filter out uppercase letters from the path.
+    * ``r"[^a-z0-9._]+"``: Don't filter out underscores from the path.
+
+    **Note:** Allowing uppercase letters here will not yield the desired result
+    unless lowercasing is also turned off. Otherwise all uppercase letters are
+    lowercased before filtering happens and no uppercase letters will appear
+    in the path anyway.
```

Comment:
This is a bit confusing. If I understand correctly:
* If you want uppercase, set `slugify-lowercase=False` and `slugify-regex="[^a-zA-Z0-9]+"`
* If you want lowercase, set `slugify-lowercase=True` and `slugify-regex="[^a-z0-9]+"`

I would be nice to somehow ensure that these can't be inconsistent, but I don't really have any clean ideas :\

Reply:
Maybe the `regex` part should just be about additional characters that are allowed? i.e. we construct the full regex ourselves:
* If `slugify-lowercase=False`: set the regex to `r"[^a-zA-Z0-9{extras}]"`
* If `slugify-lowercase=True`: set the regex to `r"[^a-z0-9{extras}]"`

where `extras = papis.config.getstring("slugify-regex")`. What do you think?

Reply:
Oh yeah, that's an awesome idea! I also thought about enforcing consistency there but didn't quite know how to go about it.

Reply:
Perfect! Thanks!